motr/cob.c

```diff
@@ -1845,18 +1846,42 @@ M0_INTERNAL int m0__obj_namei_send(struct m0_op_obj *oo)
 	}
 	cr->cr_cob_attr = cob_attr;
 
+	/* Skip cob lookup if obj.ob_attr.oa_pver is not empty.
+	 * We are assuming here, if pver is not empty that means we
+	 * have received pver from s3 */
+	skip_lookup = false;
+	obj = m0__obj_entity(oo->oo_oc.oc_op.op_entity);
+	if ((cr->cr_opcode == M0_EO_GETATTR) && (obj->ob_attr.oa_pver.f_container != 0 &&
+	    obj->ob_attr.oa_pver.f_key != 0)) {
+		skip_lookup = true;
+		M0_LOG(M0_ALWAYS, "pver "FID_F" is valid, lookup can be skipped",
```

Comment:
Don't use "M0_ALWAYS".
M0_DEBUG, or M0_INFO is preferred.

Reply:
I used M0_ALWAYS, because I wanted to print those logs on terminal upon motr utilities run, (m0cp/m0cat). 
So that we can show that during object creation (m0cp) pool version is available in obj->ob_attr.oa_pver ( So that it will be available to s3) The same pool version is used for object read (m0cat ) (sources are build again with this pool version) 
This is clearly visible with logs on terminal. All this logs are only for simulation and demonstration purpose. For the final patch I will remove all the M0_ALWAYS logs in final patch.